addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/misc/LocalNetworkInterface.java

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.homematic.internal.misc;
+
+import java.net.Inet6Address;
+import java.net.InetAddress;
+import java.net.NetworkInterface;
+import java.net.SocketException;
+import java.util.Enumeration;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Utility class to retrieve local network interfaces.
+ *
+ * @author Gerhard Riegler - Initial contribution
+ */
+public class LocalNetworkInterface {
+    private static final Logger logger = LoggerFactory.getLogger(LocalNetworkInterface.class);
+
+    /**
+     * Finds the (non loopback, non localhost) local network interface.
+     */
+    public static String getLocalNetworkInterface() {
+        try {
+            String localInterface = null;
+            Enumeration<NetworkInterface> interfaces = NetworkInterface.getNetworkInterfaces();
+            while (interfaces.hasMoreElements()) {
+                NetworkInterface current = interfaces.nextElement();
+                if (!current.isUp() || current.isLoopback() || current.isVirtual()) {
+                    continue;
+                }
+                Enumeration<InetAddress> addresses = current.getInetAddresses();
+                while (addresses.hasMoreElements()) {
+                    InetAddress current_addr = addresses.nextElement();
+                    if (current_addr.isLoopbackAddress() || (current_addr instanceof Inet6Address)) {
+                        continue;
+                    }
+                    if (localInterface != null) {
+                        logger.warn("Found multiple local interfaces! Replacing " + localInterface + " with "
```

Comment:
It tries to find the hostaddress of the server, if there are multiple network interfaces, this is logged with a warning and i am using the last one. I don't know here which interface is the correct one. This address is used by the CCU to communicate with openHab.